File: candy.py
class Solution:
    def candy(self,ratings):
        length=len(ratings)
        cNum=[1 for i in range(length)]
        for i in range(1,length):
            if ratings[i]>ratings[i-1]:
                cNum[i]=cNum[i-1]+1
        for i in range(length-2,-1,-1):
            if ratings[i]>ratings[i+1] and cNum[i]<=cNum[i+1]:
                cNum[i]=cNum[i+1]+1
        return sum(cNum)
if __name__=='__main__':
    sol=Solution()
    ratings=[1,4,3,6,2]
    print sol.candy(ratings)

File: Algorithm/CPP/candy.cpp
# include <stdio.h>
# include <iostream>
# include <vector>
# include <string>
# include <unordered_map>
# include <cctype>
# include <stack>
# include <algorithm>
# include <time.h>
# include<random>
# include <queue>
#include <stdint.h>

using namespace std;

class Solution {
public:
    int candy(vector<int>& ratings) {
		int n = ratings.size();
		vector<int> nums(n,1);
		for (int i = 1; i < n; i++)
			if (ratings[i]>ratings[i - 1]) nums[i] = nums[i - 1] + 1;
		for (int i = n - 2; i >= 0; i--)
			if (ratings[i] > ratings[i + 1] && nums[i + 1] >= nums[i])
				nums[i] = nums[i + 1] + 1;
		int sum=0;
		for (int i = 0; i < n; i++)
		{
			sum += nums[i];
		}
		return sum;        
    }
};
